app/src/main/java/com/example/android/roomwordssample/WordViewModel.kt

```diff
@@ -28,20 +29,15 @@ import kotlinx.coroutines.launch
  * an up-to-date list of all words.
  */
 
-class WordViewModel(application: Application) : AndroidViewModel(application) {
+class WordViewModel(private val repository: WordRepository) : ViewModel() {
 
-    private val repository: WordRepository
-    // Using LiveData and caching what getAlphabetizedWords returns has several benefits:
+    // Using LiveData and caching what allWords returns has several benefits:
     // - We can put an observer on the data (instead of polling for changes) and only update the
     //   the UI when the data actually changes.
     // - Repository is completely separated from the UI through the ViewModel.
-    val allWords: LiveData<List<Word>>
-
-    init {
-        val wordsDao = WordRoomDatabase.getDatabase(application, viewModelScope).wordDao()
-        repository = WordRepository(wordsDao)
-        allWords = repository.allWords
-    }
+    val allWords: LiveData<List<Word>> = repository.allWords.asLiveData(
+        viewModelScope.coroutineContext
```

Comment:
`viewModelScope.coroutineContext` is not really needed. This is needed if you want the flow to be cancelled as soon as the ViewModel is destroyed. But it's going to do it anyway after the timeout. Remove if you want to simplify the code, leave it if it makes more sense to you.